modules/rostests/apitests/shell32/SHGetComputerDisplayNameW.cpp

```diff
@@ -0,0 +1,165 @@
+/*
+ * PROJECT:     ReactOS API Tests
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Tests for SHGetComputerDisplayNameW
+ * COPYRIGHT:   Copyright 2025 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ */
+
+#include "shelltest.h"
+#include <lmserver.h>
+#include <undocshell.h>
+#include <strsafe.h>
+#include <versionhelpers.h>
+
+typedef HRESULT (WINAPI *FN_SHGetComputerDisplayNameW)(LPWSTR, DWORD, LPWSTR, DWORD);
+typedef NET_API_STATUS (WINAPI *FN_NetServerGetInfo)(LPWSTR, DWORD, PBYTE*);
+typedef NET_API_STATUS (WINAPI *FN_NetApiBufferFree)(PVOID);
+
+static FN_SHGetComputerDisplayNameW s_pSHGetComputerDisplayNameW = NULL;
+static FN_NetServerGetInfo s_pNetServerGetInfo = NULL;
+static FN_NetApiBufferFree s_pNetApiBufferFree = NULL;
+
+#define COMPUTER_DESCRIPTIONS_KEY \
+    L"Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\ComputerDescriptions"
+
+static PWSTR
+SHELL_SkipServerSlashes(
+    _In_ PCWSTR pszPath)
+{
+    PCWSTR pch;
+    for (pch = pszPath; *pch == L'\\'; ++pch)
+        ;
+    return const_cast<PWSTR>(pch);
+}
+
+static VOID
+SHELL_CacheComputerDescription(
+    _In_ PCWSTR pszServerName,
+    _In_ PCWSTR pszDesc)
+{
+    if (!pszDesc)
+        return;
+
+    DWORD cbDesc = (lstrlenW(pszDesc) + 1) * sizeof(WCHAR);
+    SHSetValueW(HKEY_CURRENT_USER, COMPUTER_DESCRIPTIONS_KEY,
+                SHELL_SkipServerSlashes(pszServerName), REG_SZ, pszDesc, cbDesc);
+}
+
+static HRESULT
+SHELL_GetCachedComputerDescription(
+    _Out_ PWSTR pszDesc,
+    _In_ DWORD cchDescMax,
+    _In_ PCWSTR pszServerName)
+{
+    cchDescMax *= sizeof(WCHAR);
+
+    LSTATUS error = SHGetValueW(HKEY_CURRENT_USER, COMPUTER_DESCRIPTIONS_KEY,
+                                SHELL_SkipServerSlashes(pszServerName), NULL, pszDesc, &cchDescMax);
+    return HRESULT_FROM_WIN32(error);
+}
+
+static HRESULT
+SHELL_BuildDisplayMachineName(
+    _Out_ PWSTR pszName,
+    _In_ DWORD cchNameMax,
+    _In_ PCWSTR pszServerName,
+    _In_ PCWSTR pszDescription)
+{
+    if (!pszDescription || !*pszDescription)
+        return E_FAIL;
+
+    PCWSTR pszFormat = SHRestricted(REST_ALLOWCOMMENTTOGGLE) ? L"%1 (%2)" : L"%2 (%1)";
+    PCWSTR args[] = { SHELL_SkipServerSlashes(pszServerName), pszDescription };
```

Comment:
I would suggest you just use the very same code that you use in the implementation in shell32.

Reply:
Fixed.